### PR TITLE
ENG-15834: Adding p3dn support

### PR DESCRIPTION
**Pull request by NatahlieWang**
#### Reviewers! 
Here's [How to do a Code Review](https://cloudhealthtech.atlassian.net/wiki/spaces/EN/pages/17989636/Code+Review)

#### Deploy readiness:

- [ ]  2 Reviews: 1 peer/buddy and 1 SME from [here](https://cloudhealthtech.atlassian.net/wiki/spaces/EN/pages/17989636/Code+Review)
- [ ]  Did you add the JIRA ticket ID to the PR title? 
- [ ]  Is your PR title **meaningful?** `[ENG-3333] Update pull request template`, not `Typo Fix` or `Add test`
- [ ]  After commits are squashed, will your primary commit message be meaningful?

#### Documentation links:

- JIRA Ticket(s):https://cloudhealthtech.atlassian.net/browse/ENG-15834
- Design Doc (where applicable):
- Link to passing Jenkins build (if PR tests are failing due to inter-repo dependencies):
  - [ ]  Branch pointers in Gemfile removed?

#### Proposed changes:
-Add support for the p3dn.24xlarge instance type, update prices for it.

#### Test Plan and Risk Assessment:
_Will this change impact other teams' code? If so, how?_

-No impact

_Test Plan:_

-Tested profile creation, price update locally

_Rollback Plan:_

### CODE REVIEW

File: lib/amazon-pricing/definitions/instance-type.rb [hunk 1]
```diff
@@ -207,15 +207,15 @@ def self.get_values(json, category_type, override_price = false)
...
       'i2.xlarge' => 'High I/O Extra Large', 'i2.2xlarge' => 'High I/O Double Extra Large', 'i2.4xlarge' => 'High I/O Quadruple Extra Large', 'i2.8xlarge' => 'High I/O Eight Extra Large',
       'i3.large' => 'Storage Optimized High I/O Large',
       'i3.xlarge' => 'Storage Optimized High I/O Extra Large', 'i3.2xlarge' => 'Storage Optimized High I/O Double Extra Large',
       'i3.4xlarge' => 'Storage Optimized High I/O Quadruple Extra Large', 'i3.8xlarge' => 'Storage Optimized High I/O Extra Large',
       'i3.16xlarge' => 'Storage Optimized High I/O Hextuple Extra Large',
       'i3.metal' => 'Storage Optimized High I/O Metal',
-      'i3p.16xlarge' => 'Storage Optimized VMware High I/O Hextuple Extra Large',
-      'd2.xlarge' => 'Dense Storage Extra Large', 'd2.2xlarge' => 'Dense Storage Double Extra Large', 'd2.4xlarge' => 'Dense Storage Quadruple Extra Large', 'd2.8xlarge' => 'Dense Storage Eight Extra Large',
+       'd2.xlarge' => 'Dense Storage Extra Large', 'd2.2xlarge' => 'Dense Storage Double Extra Large', 'd2.4xlarge' => 'Dense Storage Quadruple Extra Large', 'd2.8xlarge' => 'Dense Storage Eight Extra Large',
```
Comment (xhanshawn) [inline lib/amazon-pricing/definitions/instance-type.rb]: don't move

File: lib/amazon-pricing/definitions/instance-type.rb [hunk 2]
```diff
@@ -207,14 +207,14 @@ def self.get_values(json, category_type, override_price = false)
...
       'p3.2xlarge' => 'GPU-3 Compute Double Extra Large', 'p3.8xlarge' => 'GPU-3 Compute Eight Extra Large', 'p3.16xlarge' => 'GPU-3 Compute Hextuple Extra Large',
+      'p3dn.24xlarge' => 'GPU-3 Nvidia Twenty Four Extra Large',
       'c3.large' => 'High-Compute Large', 'c3.xlarge' => 'High-Compute Extra Large', 'c3.2xlarge' => 'High-Compute Double Extra Large', 'c3.4xlarge' => 'High-Compute Quadruple Extra Large', 'c3.8xlarge' => 'High-Compute Eight Extra Large',
       'i2.xlarge' => 'High I/O Extra Large', 'i2.2xlarge' => 'High I/O Double Extra Large', 'i2.4xlarge' => 'High I/O Quadruple Extra Large', 'i2.8xlarge' => 'High I/O Eight Extra Large',
       'i3.large' => 'Storage Optimized High I/O Large',
       'i3.xlarge' => 'Storage Optimized High I/O Extra Large', 'i3.2xlarge' => 'Storage Optimized High I/O Double Extra Large',
       'i3.4xlarge' => 'Storage Optimized High I/O Quadruple Extra Large', 'i3.8xlarge' => 'Storage Optimized High I/O Extra Large',
       'i3.16xlarge' => 'Storage Optimized High I/O Hextuple Extra Large',
       'i3.metal' => 'Storage Optimized High I/O Metal',
-      'i3p.16xlarge' => 'Storage Optimized VMware High I/O Hextuple Extra Large',
```
Comment (stanmluke) [inline lib/amazon-pricing/definitions/instance-type.rb]: just worried about a bad merge here (causing this line to be deleted)

File: lib/amazon-pricing/definitions/instance-type.rb [hunk 3]
```diff
@@ -275,7 +275,7 @@ def self.get_values(json, category_type, override_price = false)
       'db.t1.micro' => 160,
       'c3.large' => 32, 'c3.xlarge' => 80, 'c3.2xlarge' => 160, 'c3.4xlarge' => 320, 'c3.8xlarge' => 640,
       'i2.xlarge' => 800, 'i2.2xlarge' => 1600, 'i2.4xlarge' => 3200, 'i2.8xlarge' => 6400,
-      'i3.large' => 475, 'i3.xlarge' => 950, 'i3.2xlarge' => 1900, 'i3.4xlarge' => 3800, 'i3.8xlarge' => 7600, 'i3.16xlarge' => 15200, 'i3.metal' => 15200, 'i3p.16xlarge' => 15200,
+      'i3.large' => 475, 'i3.xlarge' => 950, 'i3.2xlarge' => 1900, 'i3.4xlarge' => 3800, 'i3.8xlarge' => 7600, 'i3.16xlarge' => 15200, 'i3.metal' => 15200,
```
Comment (stanmluke) [inline lib/amazon-pricing/definitions/instance-type.rb]: maybe rebase so bad merge doesnt happen?

File: lib/amazon-pricing/definitions/instance-type.rb [hunk 4]
```diff
@@ -320,7 +321,7 @@ def self.get_values(json, category_type, override_price = false)
       'db.t1.micro' => 64,
       'c3.large' => 64, 'c3.xlarge' => 64, 'c3.2xlarge' => 64, 'c3.4xlarge' => 64, 'c3.8xlarge' => 64,
       'i2.large' => 64, 'i2.xlarge' => 64, 'i2.2xlarge' => 64, 'i2.4xlarge' => 64, 'i2.8xlarge' => 64,
-      'i3.large' => 64, 'i3.xlarge' => 64, 'i3.2xlarge' => 64, 'i3.4xlarge' => 64, 'i3.8xlarge' => 64, 'i3.16xlarge' => 64, 'i3.metal' => 64, 'i3p.16xlarge' => 64,
+      'i3.large' => 64, 'i3.xlarge' => 64, 'i3.2xlarge' => 64, 'i3.4xlarge' => 64, 'i3.8xlarge' => 64, 'i3.16xlarge' => 64, 'i3.metal' => 64,
```
Comment (stanmluke) [inline lib/amazon-pricing/definitions/instance-type.rb]: rebase?

File: lib/amazon-pricing/definitions/instance-type.rb [hunk 5]
```diff
@@ -365,7 +367,7 @@ def self.get_values(json, category_type, override_price = false)
       'db.t1.micro' => :ebs,
       'c3.large' => :ssd, 'c3.xlarge' => :ssd, 'c3.2xlarge' => :ssd, 'c3.4xlarge' => :ssd, 'c3.8xlarge' => :ssd,
       'i2.large' => :ssd, 'i2.xlarge' => :ssd, 'i2.2xlarge' => :ssd, 'i2.4xlarge' => :ssd, 'i2.8xlarge' => :ssd,
-      'i3.large' => :ssd, 'i3.xlarge' => :ssd, 'i3.2xlarge' => :ssd, 'i3.4xlarge' => :ssd, 'i3.8xlarge' => :ssd, 'i3.16xlarge' => :ssd, 'i3.metal' => :ssd, 'i3p.16xlarge' => :ssd,
+      'i3.large' => :ssd, 'i3.xlarge' => :ssd, 'i3.2xlarge' => :ssd, 'i3.4xlarge' => :ssd, 'i3.8xlarge' => :ssd, 'i3.16xlarge' => :ssd, 'i3.metal' => :ssd,
```
Comment (stanmluke) [inline lib/amazon-pricing/definitions/instance-type.rb]: rebase?

**Pull request merged**